### ISSUE TITLE
Model files throw 42000 errors

### ISSUE THREAD

**Issue by curious-bryan** [jacobm001/dbt-mssql#7]
### Summary

- Trying to recreate jaffle_shop tutorial for dbt at https://github.com/fishtown-analytics/jaffle_shop
- Related to https://github.com/jacobm001/dbt-mssql/issues/6#issue-479754997
- Ran into `42000` database errors after running `dbt run`

### What happened:

- Ran into issue with loading CSVs with `dbt seed`; see https://github.com/jacobm001/dbt-mssql/issues/6#issue-479754997
- Loaded CSVs as flat files directly to DB via SQL Server Management Studio
- Ran `dbt run` <- ERRORs thrown for all 3 staging models in `jaffle_shop` `model` directory.  See errors below.

### What I expected to happen:

After loading CSVs directly into SQL Server DB, I expected `dbt run` to successfully load and run models found in `staging` and `marts`.

### Errors

```
Running with dbt=0.14.0
Found 8 models, 20 tests, 0 snapshots, 0 analyses, 113 macros, 0 operations, 3 seed files, 0 sources

11:28:24 | Concurrency: 1 threads (target='dev')
11:28:24 |
11:28:24 | 1 of 8 START view model dbt_dev.stg_orders....................... [RUN]
11:28:24 | 1 of 8 ERROR creating view model dbt_dev.stg_orders.............. [ERROR in 0.11s]
11:28:24 | 2 of 8 START view model dbt_dev.stg_payments..................... [RUN]
11:28:24 | 2 of 8 ERROR creating view model dbt_dev.stg_payments............ [ERROR in 0.04s]
11:28:24 | 3 of 8 START view model dbt_dev.stg_customers.................... [RUN]
11:28:25 | 3 of 8 ERROR creating view model dbt_dev.stg_customers........... [ERROR in 0.04s]
11:28:25 | 4 of 8 SKIP relation dbt_dev.customer_orders..................... [SKIP]
11:28:25 | 5 of 8 SKIP relation dbt_dev.customer_payments................... [SKIP]
11:28:25 | 6 of 8 SKIP relation dbt_dev.order_payments...................... [SKIP]
11:28:25 | 7 of 8 SKIP relation dbt_dev.dim_customers....................... [SKIP]
11:28:25 | 8 of 8 SKIP relation dbt_dev.fct_orders.......................... [SKIP]
11:28:25 |
11:28:25 | Finished running 3 view models, 5 table models in 1.37s.

Completed with 3 errors:

Database Error in model stg_orders (models\staging\stg_orders.sql)
  ('42000', "[42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Incorrect syntax near the keyword 'with'. (156) (SQLExecDirectW); [42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Incorrect syntax near the keyword 'with'. If this statement is a common table expression, an xmlnamespaces clause or a change tracking context clause, the previous statement must be terminated with a semicolon. (319); [42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Incorrect syntax near ','. (102); [42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Incorrect syntax near ')'. (102)")
  compiled SQL at target\compiled\jaffle_shop\staging\stg_orders.sql

Database Error in model stg_payments (models\staging\stg_payments.sql)
  ('42000', "[42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Incorrect syntax near the keyword 'with'. (156) (SQLExecDirectW); [42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Incorrect syntax near the keyword 'with'. If this statement is a common table expression, an xmlnamespaces clause or a change tracking context clause, the previous statement must be terminated with a semicolon. (319); [42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Incorrect syntax near ','. (102); [42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Incorrect syntax near ')'. (102)")
  compiled SQL at target\compiled\jaffle_shop\staging\stg_payments.sql

Database Error in model stg_customers (models\staging\stg_customers.sql)
  ('42000', "[42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Incorrect syntax near the keyword 'with'. (156) (SQLExecDirectW); [42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Incorrect syntax near the keyword 'with'. If this statement is a common table expression, an xmlnamespaces clause or a change tracking context clause, the previous statement must be terminated with a semicolon. (319); [42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Incorrect syntax near ','. (102); [42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Incorrect syntax near ')'. (102)")
  compiled SQL at target\compiled\jaffle_shop\staging\stg_customers.sql

Done. PASS=0 ERROR=3 SKIP=5 TOTAL=8
```

### Environment Details

- See https://github.com/jacobm001/dbt-mssql/issues/6#issue-479754997
- Let me know if you need more details

**Comment by curious-bryan** [comment]
@jacobm001, heads up.  I think I might have spotted the issue.  When you get from traveling for work, you may want to check out Lines 64-69 in `dbt-mssql/dbt/include/mssql/macros/adapters.sql` (https://github.com/jacobm001/dbt-mssql/blob/master/dbt/include/mssql/macros/adapters.sql).

`dbt-mssql` (`1.0.3`) lists 

```
{% macro mssql__create_table_as(temporary, relation, sql) -%}
    with cte as (
      {{ sql }}  
    ) select * into {{ relation.schema }}.{% if temporary: -%}#{%- endif %}{{ relation.identifier }}
    from cte;
{% endmacro %}
```
But I did not see any reference to `cte`.  I think this with statement might be causing the `42000` error when creating the table.

Perhaps, a suggested edit could be something like

```
{% macro mssql__create_table_as(temporary, relation, sql) -%}
	SELECT * 
	INTO {% if temporary: -%}#{%- endif %}{{ relation.identifier }}
	FROM {{ sql }}
	;
{% endmacro %}
```
Thanks for putting time into this adapter!  Pumped about next update!

**Comment by jacobm001** [comment]
Looks like this issue is being caused by the `mssql__create_view_as` macro. It works fine if the sql it's provided does not contain a CTE. If it does, sql server considers it a syntax error.

Simply removing the parenthesis wrapper after the `as` keyword appears to resolve this issue. I will make that alteration and report back after pushing it through the test harness.

**Comment by jacobm001** [comment]
I've fixed this issue in the latest version. The issue ended up being caused both by the `create view` and `create table as` macros. I've fixed the later with some inspiration from the `dbt-sqlserver` package.

Unfortunately, Jaffle Shop contains some elements that I can't support with this package. SQL Server supports neither the using, nor the group by ordinal syntax. I'll add that to the readme, but I'm not sure if it's worth creating a fork of jaffle shop or not.
